types/proposal.go

```diff
@@ -89,11 +90,19 @@ func (p *Proposal) ValidateBasic() error {
 //
 // For more information on the meaning of 'timely', see the proposer-based timestamp specification:
 // https://github.com/tendermint/spec/tree/master/spec/consensus/proposer-based-timestamp
-func (p *Proposal) IsTimely(recvTime time.Time, sp SynchronyParams) bool {
+func (p *Proposal) IsTimely(recvTime time.Time, sp SynchronyParams, round int32) bool {
+	// The message delay values are scaled as rounds progress.
+	// Every 10 rounds, the message delay is doubled to allow consensus to
```

Comment:
Should we set a cap on this, e.g., after x0 rounds we cap out at 2^x times the original delay? Otherwise that exponential growth could lead to a Rip van Winkle situation (the circumstances of that would be weird, but "sleep for an hour" doesn't make it easier to debug)

Reply:
Ah, so `MessageDelay` does not impact how long a node will end up sleeping for at any point so I don't think that poses much risk unless I'm misunderstanding. However, we are at risk I'm realizing of integer overflow if the round gets too large. I've added a check to handle this that just uses the max duration when the value computed is no longer positive.